### ISSUE TITLE
DroneCore application (GCS) should NOT mix up with System/Device

### ISSUE THREAD

**Issue by shakthi-prashanth-m** [dronecore/DroneCore#317]
class `Device` (we're going to rename it to `System`) represents a MAVLink based system (robot/vehicle/stand alone camera, etc). On the other side, we've GCS (i.e DroneCore application). These two are different entities. I think, we shouldn't mix them up.

**Why does `Device` class hold GCS info ?**
https://github.com/dronecore/DroneCore/blob/ca93a9cfb01d57234e09c5ac22d3261caa2105c5/core/device.h#L169-L171

IMO, we should identify GCS different from `Device` class and have something like this:

```cpp
// This represents GCS DroneCore client
struct Self {
static constexpr uint8_t system_id = 0;
static constexpr uint8_t component_id = MAV_COMP_ID_SYSTEM_CONTROL;
static constexpr MAV_TYPE type = MAV_TYPE_GCS;
};
```

And, use them as follows:

```cpp
mavlink_msg_command_long_pack(Self::system_id, /* <= here */
Self::component_id, /* <= here */
&new_work.mavlink_message,
target_system_id,
target_component_id,
command,
0,
params.v[0], params.v[1], params.v[2], params.v[3],
params.v[4], params.v[5], params.v[6]);
...
```

Does it make sense ?

**Comment by julianoes** [comment]
That's correct. Is this currently implemented wrongly?

**Comment by shakthi-prashanth-m** [comment]
Yes, I think its wrong.

**Comment by shakthi-prashanth-m** [comment]
Addressed in #310.
